provider.go

```diff
@@ -172,21 +167,20 @@ func GetCredential(secrets map[string]string) (string, string, error) {
 }
 
 func (p *Provider) getVaultURL(ctx context.Context, cloudName string) (vaultURL *string, err error) {
-	log.Debugf("subscriptionID: %s", p.SubscriptionID)
 	log.Debugf("vaultName: %s", p.KeyvaultName)
-	log.Debugf("resourceGroup: %s", p.ResourceGroup)
 
-	vaultsClient := kvmgmt.NewVaultsClient(p.SubscriptionID)
-	token, tokenErr := p.GetManagementToken(AuthGrantType(), cloudName)
-	if tokenErr != nil {
-		return nil, errors.Wrapf(tokenErr, "failed to get management token")
+	// See docs for validation spec: https://docs.microsoft.com/en-us/azure/key-vault/about-keys-secrets-and-certificates#objects-identifiers-and-versioning
+	if match, _ := regexp.MatchString("[-a-zA-Z0-9]{3,24}", p.KeyvaultName); !match {
```

Comment:
can we also add a unit test for `getVaultURL` to test for expected string pattern for the keyvault name and the vaultUri

Reply:
sure thing! Will add that.

Reply:
done, PTAL!